python/pyspark/serializers.py

```diff
@@ -382,18 +382,30 @@ def _hijack_namedtuple():
         return
 
     global _old_namedtuple  # or it will put in closure
+    global _old_namedtuple_kwdefaults  # or it will put in closure too
 
     def _copy_func(f):
         return types.FunctionType(f.__code__, f.__globals__, f.__name__,
                                   f.__defaults__, f.__closure__)
 
+    def _kwdefaults(f):
+        kargs = getattr(f, "__kwdefaults__", None)
```

Comment:
after applying patch can you try to run 
sc.parallelize(range(100), 8)
and confirm that it is working, because it is not...
and serialisation of objects goes crazy.. 

Thanks for your efforts